c/httpserver.c

```diff
@@ -4053,11 +4053,31 @@ void respondWithUnixFile2(HttpService* service, HttpResponse* response, char* ab
 #error Unknown OS
 #endif
         ;
-      if (ccsid == 0) {
-        streamTextForFile2(response, NULL, in, ENCODING_CHUNKED, NATIVE_CODEPAGE, webCodePage, asB64);
-      } else {
-        streamTextForFile2(response, NULL, in, ENCODING_CHUNKED, ccsid, webCodePage, asB64);
-      }
+    char *forceEnabled = getQueryParam(response->request, "force");
+    if (ccsid == 0 && strcmp(forceEnabled, "enable")) {
+        char *sourceEncoding = getQueryParam(response->request, "source");
+        char *targetEncoding = getQueryParam(response->request, "target");
+        int sEncoding;
+        int tEncoding;
+        if(sourceEncoding != NULL && targetEncoding != NULL) {
+           int sscanfSource = sscanf(sourceEncoding, "%d", &sEncoding);
+           int sscanfTarget = sscanf(targetEncoding, "%d", &tEncoding);
+           if (sscanfSource != 1 || sscanfTarget != 1) {
+             respondWithError(response, HTTP_STATUS_BAD_REQUEST, "source/target encoding value parsing error");
+             return;
+           }
+           streamTextForFile2(response, NULL, in, ENCODING_CHUNKED, sEncoding, tEncoding, asB64);
+        }
+        else {
+           streamTextForFile2(response, NULL, in, ENCODING_CHUNKED, NATIVE_CODEPAGE, webCodePage, asB64);
```

Comment:
Is this expected behavior to continue with this call implicitly when either of the encodings hasn't been specified? Would it be better to return `HTTP_STATUS_BAD_REQUEST` in the following case
```c
       if(sourceEncoding != NULL && targetEncoding != NULL) {
           int sscanfSource = sscanf(sourceEncoding, "%d", &sEncoding);
           int sscanfTarget = sscanf(targetEncoding, "%d", &tEncoding);
           if (sscanfSource != 1 || sscanfTarget != 1) {
             respondWithError(response, HTTP_STATUS_BAD_REQUEST, "source/target encoding value parsing error");
             return;
           }
           streamTextForFile2(response, NULL, in, ENCODING_CHUNKED, sEncoding, tEncoding, asB64);
        }
        else {
          respondWithError(response, HTTP_STATUS_BAD_REQUEST, "source and target encodings are required");
          return;
        }
```

Reply:
Hm I hope so becuase the request doesn't content required params ? 
@ifakhrutdinov

Reply:
I think if there was an intent to force the encoding but not enough parameters have been provided, it'd be better to return a bad request error. Otherwise we'll do something that wasn't intended.

Reply:
sure I will do it.

Reply:
You should consider `strtol()` instead of `sscanf()`. It has better error checking and is the canonical way of converting a string to an integer.